superset/viz.py

```diff
@@ -474,6 +475,18 @@ def get_payload(self, query_obj: Optional[QueryObjectDict] = None) -> VizPayload
         filters = self.form_data.get("filters", [])
         filter_columns = [flt.get("col") for flt in filters]
         columns = set(self.datasource.column_names)
+        filter_values_columns = []
+
+        # if using virtual datasource, check filter_values
+        sql = self.datasource.sql
```

Comment:
Note I don't think this indirection is necessary.

Reply:
fixed.